### PR TITLE
Byte converter update

### PR DESCRIPTION
**Pull request by tnine**
This is the merge with the 0.7 functionality, as well an update to the Bytes class to support Serializable classes.  This is required for my Cassandra plugin, where FK's are persisted as the serializable identity class.

### CODE REVIEW

**Comment by danwashusen** [comment]
Indexed stuff looks good.

Not 100% sure I understand the need to add java serialization helpers to the Bytes class.  You could have these helpers methods in your Cassandra plugin code couldn't you?

**Comment by tnine** [comment]
You're correct, I could.  However serializing a Serializable instance is a pretty common thing.  It was the only piece of functionality I'm missing from Pelops now, and it's still pretty generic.

**Comment by danwashusen** [comment]
Yep, serialization is a pretty common problem but I don't think Pelops needs to be aware of it.  There are also many options available (http://github.com/eishay/jvm-serializers/wiki) when it comes to serialization and most are arguably better than Java's inbuilt functionality.   The application that my company is working on uses Jackson for example and Dominic's company (from memory) uses XStream.

So in short, I don't think Pelops needs to be aware of it...

**Comment by tnine** [comment]
Removed serialization
